cognitive/src/main/scala/com/microsoft/azure/synapse/ml/services/openai/OpenAIPrompt.scala

```diff
@@ -62,60 +63,117 @@ class OpenAIPrompt(override val uid: String) extends Transformer
     set(postProcessingOptions, v.asScala.toMap)
 
   val dropPrompt = new BooleanParam(
-    this, "dropPrompt", "whether to drop the column of prompts after templating")
+    this, "dropPrompt", "whether to drop the column of prompts after templating (when using legacy models)")
 
   def getDropPrompt: Boolean = $(dropPrompt)
 
   def setDropPrompt(value: Boolean): this.type = set(dropPrompt, value)
 
+  val dropMessages = new BooleanParam(
+    this, "dropMessages", "whether to drop the column of messages after templating (when using gpt-4 or higher)")
+
+  def getDropMessages: Boolean = $(dropMessages)
+
+  def setDropMessages(value: Boolean): this.type = set(dropMessages, value)
+
```

Comment:
can we just re-use dropPrompt to choose whether to drop either the prompt or messages column?